src/libraries/System.Diagnostics.DiagnosticSource/tests/MetricsTests.cs

```diff
@@ -807,6 +807,94 @@ public void SerializedEventsTest()
             }).Dispose();
         }
 
+        [ConditionalFact(typeof(RemoteExecutor), nameof(RemoteExecutor.IsSupported))]
+        public void TestRecordingMeasurementsWithTagList()
+        {
+            RemoteExecutor.Invoke(() => {
+
+                Meter meter = new Meter("RecordingMeasurementsWithTagList");
+
+                using (MeterListener listener = new MeterListener())
+                {
+                    Counter<int> counter = meter.CreateCounter<int>("Counter");
+                    Histogram<int> histogram = meter.CreateHistogram<int>("histogram");
+
+                    listener.EnableMeasurementEvents(counter, counter);
+                    listener.EnableMeasurementEvents(histogram, histogram);
+
+                    KeyValuePair<string, object?>[] expectedTags = null;
+
+                    listener.SetMeasurementEventCallback<int>((inst, measurement, tags, state) => {
+                        for (int i = 0; i < expectedTags.Length; i++)
+                        {
+                            Assert.Equal(expectedTags[i], tags[i]);
+                        }
+                    });
+
+                    // 0 Tags
+
+                    expectedTags = new KeyValuePair<string, object?>[0];
+                    counter.Add(10, new TagList());
+                    histogram.Record(10, new TagList());
+
+                    // 1 Tags
+                    expectedTags = new KeyValuePair<string, object?>[] { new KeyValuePair<string, object?>("Key1", "Value1") };
+                    counter.Add(10, new TagList() { expectedTags[0] });
+                    histogram.Record(10, new TagList() { new KeyValuePair<string, object?>("Key1", "Value1") });
+
+                    // 2 Tags
+                    expectedTags = new KeyValuePair<string, object?>[]
```

Comment:
Is there a way to have some syntactic sugar to make it shorter/simpler?
e.g. ideally something like:

```csharp
new KeyValuePair<string, object?>[] {
    ["Key1"] = "Value1",
    ["Key2"] = "Value2",
}
```

Reply:
I think it is possible.